Python/Product/Analysis/Intellisense/ImportedModuleNameWalker.cs

```diff
@@ -18,41 +18,83 @@
 using System.Linq;
 using Microsoft.PythonTools.Analysis;
 using Microsoft.PythonTools.Analysis.Infrastructure;
+using Microsoft.PythonTools.Analysis.LanguageServer;
 using Microsoft.PythonTools.Parsing.Ast;
 
 namespace Microsoft.PythonTools.Intellisense {
+    sealed class NamedLocation {
+        public string Name { get; set; }
+        public Range Range { get; set; }
+    }
+
     class ImportedModuleNameWalker : PythonWalkerWithLocation {
         private readonly string _importingFromModuleName;
         private readonly string _importingFromFilePath;
+        private readonly PythonAst _ast;
 
-        public ImportedModuleNameWalker(IPythonProjectEntry entry, int location) : 
-            this(entry.ModuleName, entry.FilePath, location) {
+        public ImportedModuleNameWalker(IPythonProjectEntry entry, int location, PythonAst ast) :
+            this(entry.ModuleName, entry.FilePath, location, ast) {
         }
-        public ImportedModuleNameWalker(string importingFromModuleName, string importingFromFilePath, int location) : base(location) {
+        public ImportedModuleNameWalker(string importingFromModuleName, string importingFromFilePath, int location, PythonAst ast) : base(location) {
             _importingFromModuleName = importingFromModuleName;
             _importingFromFilePath = importingFromFilePath;
+            _ast = ast;
         }
 
-        public IEnumerable<string> ImportedModules { get; private set; } = Enumerable.Empty<string>();
+        public IEnumerable<NamedLocation> ImportedModules { get; private set; } = Enumerable.Empty<NamedLocation>();
+        public IEnumerable<NamedLocation> ImportedMembers { get; private set; } = Enumerable.Empty<NamedLocation>();
+        public NamedLocation ImportedType { get; private set; }
 
         public override bool Walk(FromImportStatement node) {
             if (node.StartIndex <= Location && Location <= node.EndIndex) {
```

Comment:
Also change this to check against `node.Root.StartIndex` and EndIndex, so we only get the tooltip when hovering over the module name and not the other imported names.

Reply:
OK

Reply:
Actually, it is an entry point for the entire node, then code determines where exactly it is - i.e. in `from os import path as P` it finds if hover is over `os`, `path` or `P`.

Reply:
You'll want to merge from master and undo my other fix for this then. But do make sure that the actual Hover code uses the normal handling when the user hovers over an actual variable in the file (`P` in your example). We certainly shouldn't be displaying `os` in that case, and it'll be most consistent to let the normal handling get it.